theano/tensor/opt.py

```diff
@@ -3360,6 +3360,23 @@ def local_incsubtensor_of_zeros(node):
             return
 
 
+@register_canonicalize
+@register_stabilize
+@gof.local_optimizer([IncSubtensor])
+def incsubtensor_of_zeros_to_setsubtensor(node):
+    """
+    IncSubtensor(zeros, x, ...) -> SetSubtensor(zeros, x, ...)
+
+    """
+    if (isinstance(node.op, (IncSubtensor)) and not node.op.set_instead_of_inc):
+        x = node.inputs[0]
+        y = node.inputs[1]
+
+        if isinstance(x, T.zeros):
+            node.op.set_instead_of_inc = True
```

Comment:
You should not modify an op instance. You should create a new one with the new parameter values.